services-toolkit/how-to-guides/configure-dynamic-provisioning-tanzu-postgresql.hbs.md

```diff
@@ -0,0 +1,354 @@
+        - string:
+            fmt: '%s-app-user-db-secret'
+            type: Format
+          type: string
+        type: FromCompositeFieldPath
+      - fromFieldPath: metadata.name
+        toFieldPath: spec.connectionDetails[2].name
+        transforms:
+        - string:
+            fmt: '%s-app-user-db-secret'
+            type: Format
+          type: string
+        type: FromCompositeFieldPath
+      - fromFieldPath: metadata.name
+        toFieldPath: spec.connectionDetails[3].name
+        transforms:
+        - string:
+            fmt: '%s-app-user-db-secret'
+            type: Format
+          type: string
+        type: FromCompositeFieldPath
+      - fromFieldPath: metadata.name
+        toFieldPath: spec.connectionDetails[4].name
+        transforms:
+        - string:
+            fmt: '%s-app-user-db-secret'
+            type: Format
+          type: string
+        type: FromCompositeFieldPath
+      - fromFieldPath: metadata.name
+        toFieldPath: spec.connectionDetails[5].name
+        transforms:
+        - string:
+            fmt: '%s-app-user-db-secret'
+            type: Format
+          type: string
+        type: FromCompositeFieldPath
+      - fromFieldPath: metadata.name
+        toFieldPath: spec.connectionDetails[6].name
+        transforms:
+        - string:
+            fmt: '%s-app-user-db-secret'
+            type: Format
+          type: string
+        type: FromCompositeFieldPath
+      - fromFieldPath: metadata.name
+        toFieldPath: spec.connectionDetails[7].name
+        transforms:
+        - string:
+            fmt: '%s-app-user-db-secret'
+            type: Format
+          type: string
+        type: FromCompositeFieldPath
+    readinessChecks:
+      - type: MatchString
+        fieldPath: status.atProvider.manifest.status.currentState
+        matchString: "Running"
+```
+
+This `Composition` is intended to be used as a starting point only.
+
+Once you are happy with the configuration, use `kubectl` to apply the file to the TAP cluster.
+
+```console
+$ kubectl apply -f xpostgresqlinstances.database.tanzu.example.org.composition.yml
+```
+
+## <a id="stk-dynamic-provisioning-tanzu-psql-step-5"></a> Step 5: Make the service discoverable to application teams
+
+Create a file named `tanzu-psql.class.yml` and copy in the following contents.
+
+```yaml
+# tanzu-psql.class.yml
+
+---
+apiVersion: services.apps.tanzu.vmware.com/v1alpha1
+kind: ClusterInstanceClass
+metadata:
+  name: tanzu-psql
+spec:
+  description:
+    short: VMware Tanzu Postgres
+  provisioner:
+    crossplane:
+      compositeResourceDefinition: xpostgresqlinstances.database.tanzu.example.org
+```
+
+Then use `kubectl` to apply the file to the TAP cluster.
+
+```console
+$ kubectl apply -f tanzu-psql.class.yml
+```
+
+## <a id="stk-dynamic-provisioning-tanzu-psql-step-6"></a> Step 6: Configure RBAC
+
+Create a file named `provider-kubernetes-tanzu-postgres-read-writer.rbac.yml` and copy in the following contents.
```

Comment:
**thought/non-blocking** Does TAP docs standardise on the "create a file" then k apply -f file.yaml over doing a `cat <<EOF | kubectl apply -f -` I personally get so mad having to create files like this.

Reply:
I'm not sure maybe the docs team can weigh in on this? I think that's what I've seen in other TAP docs. I have standardized on this approach for all the docs in this PR so that at least it is consistent, but don't have a strong opinion on one approach vs the other.